src/clients/KubectlClient.ts

```diff
@@ -101,10 +100,73 @@ export class KubectlClient implements IClient {
         }
     }
 
+    public async getContainersList(podName: string): Promise<string[]> {
+        if (!podName) {
+            this._logger.error(TelemetryEvent.KubectlClient_GetPodNameError, new Error(`Pod name is null`));
+            return null;
+        }
+        try {
+            // adding these decent amount of known sidecars to filter out the sidecars from the list
+            const knownSideCars: string[] = ['linkerd-proxy', 'linkerd-init', 'istio-proxy', 'darpd', 'jaeger-agent', 'nginx-proxy'];
+            const kubeconfigPath: string = await this._accountContextManager.getKubeconfigPathAsync();
+            const args: string[] = [`get`, `pod`, podName, `-o`, `jsonpath="{.spec['containers','initContainers'][*].name}"`];
```

Comment:
Is it possible to have an `initContainer` and a `container` that both have the same name? If so, I think these would be indistinguishable in the list.

Reply:
I am refactoring all this with javascript client for k8s. this will be replaced